pytorch_lightning/trainer/deprecated_api.py

```diff
@@ -153,4 +153,7 @@ def testing(self, val: bool) -> None:
         if val:
             self._running_stage = RunningStage.TESTING
         else:
-            self._running_stage = None
+            if self._running_stage == RunningStage.TRAINING:
+                pass
+            else:
+                self._running_stage = None
```

Comment:
```suggestion
            if self._running_stage != RunningStage.TRAINING:
                self._running_stage = None
```
or move even one level up as `elif`